src/sentry/monitors/endpoints/organization_monitor_details.py

```diff
@@ -156,32 +156,31 @@ def delete(self, request: Request, organization, project, monitor) -> Response:
                             MonitorStatus.DELETION_IN_PROGRESS,
                         ]
                     )
-                    .first()
                 )
             else:
-                monitor_object = (
-                    Monitor.objects.filter(id=monitor.id)
-                    .exclude(
-                        status__in=[
-                            ObjectStatus.PENDING_DELETION,
-                            ObjectStatus.DELETION_IN_PROGRESS,
-                        ]
-                    )
-                    .first()
+                monitor_objects = Monitor.objects.filter(id=monitor.id).exclude(
+                    status__in=[
+                        ObjectStatus.PENDING_DELETION,
+                        ObjectStatus.DELETION_IN_PROGRESS,
+                    ]
                 )
-            if not monitor_object or not monitor_object.update(
+
+            # create copy of queryset as update will remove objects
+            monitor_objects_list = list(monitor_objects)
+            if not monitor_objects or not monitor_objects.update(
                 status=ObjectStatus.PENDING_DELETION
             ):
                 return self.respond(status=404)
 
-            schedule = ScheduledDeletion.schedule(monitor_object, days=0, actor=request.user)
-            self.create_audit_entry(
-                request=request,
-                organization=project.organization,
-                target_object=monitor_object.id,
-                event=audit_log.get_event_id("MONITOR_REMOVE"),
-                data=monitor_object.get_audit_log_data(),
-                transaction_id=schedule.guid,
-            )
+            for monitor_object in monitor_objects_list:
+                schedule = ScheduledDeletion.schedule(monitor_object, days=0, actor=request.user)
+                self.create_audit_entry(
+                    request=request,
+                    organization=project.organization,
+                    target_object=monitor_object.id,
+                    event=audit_log.get_event_id("MONITOR_REMOVE"),
```

Comment:
It's a bit weird that we record the same event for removing a monitor and a monitor environment

Reply:
will change

Reply:
well actually the `audit_log_data` is different. but not sure if that's sufficient